src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/MethodBodyAnalyzer.cs

```diff
@@ -0,0 +1,320 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Azure.ClientSdk.Analyzers 
+{
+    internal readonly struct MethodBodyAnalyzer 
+    {
+        private readonly AsyncAnalyzerUtilities _asyncUtilities;
+        private readonly INamedTypeSymbol _boolType;
+        private readonly INamedTypeSymbol _azureTaskExtensionsType;
+        private readonly Stack<(IEnumerator<IOperation>, MethodAnalysisContext)> _symbolIteratorsStack;
+        private readonly Action<Diagnostic> _reportDiagnostic;
+
+        public MethodBodyAnalyzer(Action<Diagnostic> reportDiagnostic, Compilation compilation, AsyncAnalyzerUtilities utilities) 
+        {
+            _reportDiagnostic = reportDiagnostic;
+            _asyncUtilities = utilities;
+            
+            _boolType = compilation.GetSpecialType(SpecialType.System_Boolean);
+            _azureTaskExtensionsType = compilation.GetTypeByMetadataName("Azure.Core.Pipeline.TaskExtensions");
+            _symbolIteratorsStack = new Stack<(IEnumerator<IOperation>, MethodAnalysisContext)>();
+        }
+
+        public void Run(IMethodSymbol method, IBlockOperation methodBody) 
```

Comment:
I'm worried about how traversing every method affects the compilation time. Can you please compare Azure.Core.All compilation time with and without the analyzer enabled?